pkg/plugins/policies/meshtrace/api/v1alpha1/validator.go

```diff
@@ -0,0 +1,113 @@
+package v1alpha1
+
+import (
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/asaskevich/govalidator"
+	"golang.org/x/exp/slices"
+
+	common_proto "github.com/kumahq/kuma/api/common/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core/validators"
+	matcher_validators "github.com/kumahq/kuma/pkg/plugins/policies/matchers/validators"
+)
+
+func (r *MeshTraceResource) validate() error {
+	var verr validators.ValidationError
+	path := validators.RootedAt("spec")
+	verr.AddErrorAt(path.Field("targetRef"), validateTop(r.Spec.GetTargetRef()))
+	verr.AddErrorAt(path.Field("default"), validateDefault(r.Spec.GetDefault()))
+	return verr.OrNil()
+}
+func validateTop(targetRef *common_proto.TargetRef) validators.ValidationError {
+	targetRefErr := matcher_validators.ValidateTargetRef(targetRef, &matcher_validators.ValidateTargetRefOpts{
+		SupportedKinds: []common_proto.TargetRef_Kind{
+			common_proto.TargetRef_Mesh,
+			common_proto.TargetRef_MeshSubset,
+			common_proto.TargetRef_MeshService,
+			common_proto.TargetRef_MeshServiceSubset,
+			common_proto.TargetRef_MeshGatewayRoute,
+		},
+	})
+	return targetRefErr
+}
+
+func validateDefault(conf *MeshTrace_Conf) validators.ValidationError {
+	var verr validators.ValidationError
+
+	if conf == nil {
+		verr.AddViolation("", validators.MustBeDefined)
+		return verr
+	}
+
+	if len(conf.GetBackends()) != 1 {
+		verr.AddViolation("backends", "must have exactly one backend defined")
+	} else {
+		backend := conf.GetBackends()[0]
+		if (backend.GetDatadog() != nil) == (backend.GetZipkin() != nil) {
+			verr.AddViolation("backends[0]", validators.MustHaveOnlyOne("backend", "datadog", "zipkin"))
+		}
+
+		if backend.GetDatadog() != nil {
+			datadogBackend := backend.GetDatadog()
+			if datadogBackend.Address == "" {
+				verr.AddViolation("backends[0].datadog.address", "must not be empty")
```

Comment:
I think it's better to use `PathBuilder`, easier maintain validation when renaming fields